examples/ignition-kargs-helper

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Mount /boot. Note that we mount /boot but we don't unmount it because we
+# are run in a systemd unit with MountFlags=slave so it is unmounted for us.
+bootmnt=/mnt/boot_partition
+mkdir -p ${bootmnt}
+bootdev=/dev/disk/by-label/boot
+mount -o rw ${bootdev} ${bootmnt}
+grubcfg="${bootmnt}/grub/grub.cfg"
+
+kernelopts="$(grep kernelopts= <$grubcfg)"
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+--should-exist)
+    arg="$2"
+    # don't repeat the arg
+    if [[ ! " ${kernelopts[*]} " =~ " ${arg} " ]]; then
+        kernelopts="$kernelopts $arg "
+    fi
+    shift 2
+    ;;
+--should-not-exist)
+    arg="$2"
+    kernelopts="$(echo "$kernelopts" | sed "s|$2||g")"
+    shift 2
+    ;;
+    *)
+    echo "Unknown option"
+    exit 1
+    ;;
+esac
+done
+
+# only apply the changes & reboot if changes have been made
+if [[ ! "$kernelopts" == "$(grep kernelopts= <$grubcfg)" ]]; then
+    awk "{sub(/kernelopts=.*/,\"$kernelopts\"); print}" $grubcfg > tmp && mv tmp $grubcfg
```

Comment:
I'm not sure we can assume that we have `sed` and `awk` in the initrd.  Explicitly installing them from the dracut module would be unnecessary in the common case, so I'm not sure what to do.  Maybe just add a comment noting the dependency?

Would `sed` not work here?

Assumes we're in a writable directory.  Probably better to make `tmp` an absolute path.

Reply:
> I'm not sure we can assume that we have `sed` and `awk` in the initrd. Explicitly installing them from the dracut module would be unnecessary in the common case, so I'm not sure what to do. Maybe just add a comment noting the dependency?

:+1:

> Would `sed` not work here?

I spent some time trying to get `sed` to work and couldn't get the incantation exactly right. I can spend some more time if we want to drop the `awk` depdency.

> Assumes we're in a writable directory. Probably better to make `tmp` an absolute path.

Sure, I'll change it to `/run/kargs_tmp`

Reply:
How about `sed -i "s|^\(kernelopts=\).*|\1$kernelopts|" $grubcfg`?